### PR TITLE
add user registration flow

### PR DESCRIPTION
**Pull request by doodyparizada**
user registration flow.

There are some open questions about when does the user get the 6000 get started KIN.

### CODE REVIEW

File: scripts/src/routes/users.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export type AuthToken = {
+	token: string;
+	activated: boolean;
+}
+
+/**
+ * sign in a user
+ */
+router.post("/", async (req, res, next) => {
+	req.body.jwt;
```
Comment (nitzantomer) [inline scripts/src/routes/users.ts]: what's up with these 3 lines?

File: scripts/src/routes/users.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	req.body.device_id;
+	req.body.public_address;
+	let {userId , appId} = validateJWT(req.body.jwt); // throws if JWT not valid // XXX test this case
```
Comment (nitzantomer) [inline scripts/src/routes/users.ts]: can be `const`

File: scripts/src/routes/users.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	let user = db.Users.find({appId: appId, appUserId: userId});
```
Comment (nitzantomer) [inline scripts/src/routes/users.ts]: can be `const`

File: scripts/src/routes/users.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	if (!user) {
+		// new user
+		user = db.User.create(appId, userId, req.body.public_address, /*activated = */ false);
+		// create wallet with lumens:
+		// kin.sdk.createWallet(req.body.public_address);
+	}
+
+	let authToken = db.AuthToken.create(user.id, req.body.device_id);
```
Comment (nitzantomer) [inline scripts/src/routes/users.ts]: can be `const`

File: scripts/src/routes/users.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+	res.status(200).send({token: authToken.token, activated: user.activated});
+});
+
+/**
+ * user activates by approving TOS
+ */
+router.post("/me/activate", async (req, res, next) => {
+	const result = null; // activate user if not already activated and fund the account with KIN
```
Comment (nitzantomer) [inline scripts/src/routes/users.ts]: doesn't seem like you're using this variable.

File: scripts/src/services/users.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		key: string;
+	};
+	payload: JWTClaims & {
+		// custom claims
+		user_id: string;
+	};
+	signature: string;
+}
+
+function getApplicationPublicKey(application_id: string, key_id: string) {
```
Comment (nitzantomer) [inline scripts/src/services/users.ts]: camel case:
```
function getApplicationPublicKey(applicationId: string, keyId: string) {
```

File: scripts/src/services/users.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	// return the public key for the given application.
+	// an application might have multiple keys. each key identified by key_id.
+
+	const publicKeys = {
```
Comment (nitzantomer) [inline scripts/src/services/users.ts]: I think it's better to put this in another file and load (import) it

File: scripts/src/services/users.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+		kik: {one: "-----BEGIN PUBLIC KEY-----\n" +
+			"MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDdlatRjRjogo3WojgGHFHYLugdUWAY9iR3fy4arWNA1KoS8kVw33cJibXr8bvwUAUparCwlvdbH6dvEOfou0/gCFQsHUfQrSDv+MuSUMAe8jzKE4qW+jK+xQU9a03GUnKHkkle+Q0pX/g6jXZ7r1/xAK5Do2kQ+X5xK9cipRgEKwIDAQAB\n" +
+			"-----END PUBLIC KEY-----"}};
+
+	return publicKeys[application_id][key_id];
+}
+
+export function validateJWT(jwt: string): {userId: string, appId: string} {
+
+	/*
```
Comment (nitzantomer) [inline scripts/src/services/users.ts]: what's this comment?

File: scripts/src/services/users.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	jsonwebtoken.verify(jwt, publicKey);
+
+	return {
+		userId: decoded.payload.user_id,
+		appId: decoded.payload.iss,
+	};
+}
+
+function getUser(userId: string) {
+	return;
```
Comment (nitzantomer) [inline scripts/src/services/users.ts]: nice

File: scripts/src/services/users.ts (hunk 7 above)
Comment (doodyparizada) [reply scripts/src/services/users.ts]: for sure - it will be loaded from the DB and cached in memory

**Comment by doodyparizada** [comment]
I merging this as is. We are migrating to a new ORM which will change most of this code.

**Pull request merged**